content/posts/about_inlining.md

```diff
@@ -28,22 +28,23 @@ Source : [inline specifier – cppreference.com][1]
 
 With that in mind, let’s take a tour of the pros and the cons of this feature.
 
 ## Pros
-First of all, even if it’s a ***hint***, the keyword **still** has a chance to influence the compiler. It may not, but it also may. For instance, we can think of the stupidest compiler that will just strictly respect the `inline` keyword, without trying to analyze the context and optimize itself. This is not against the standard (according to the C++11 standard §12.1.5) and the keyword ***is*** useful in this case.
+First of all, even if it is just a ***hint***, the keyword **still** has a chance to influence the compiler. It may not, but it also may. For instance, we can think of the stupidest compiler that will just strictly respect the `inline` keyword, without trying to analyze the context and optimize itself. This is not against the standard (according to the C++11 standard §12.1.5) and the keyword ***is*** useful in this case.
+Do not get discouraged if that seems hopelessly vague. The flexibility of the above is actually a huge advantage: it lets the compiler treat large functions differently from small ones, plus it lets the compiler generate code that is easy to debug if you select the right compiler options.
 
 To be summarized, the pros are:
 
 1. It speeds up your program by avoiding function calling overhead.
-2. It saves overhead of variables' push/pop on the stack when funcion invoked.
-3. It saves overhead of return call from a function.
-4. It increases locality of reference by utilizing instruction cache.
-5. By marking it as inline, you can put a function definition in a header file (i.e. it can be included in multiple compilation unit, without the linker complaining).
+2. It saves the overhead of variables' push/pop on the stack when the function is called.
+3. It saves the overhead of return call from a function.
+4. It increases the locality of reference by utilizing instruction cache.
+5. By marking it as inline, you can put a function definition in a header file (i.e. it can be included in multiple compilation units, without the linker complaining).
 
-The first three benefits are overally the main benefits of this feature, and the original goal of the introduction of this keyword. Those who know a bit of assembly know that pushing a lot of parameters on the stack to call function can cost more instructions that the function holds.
+The first three benefits are overall the main benefits of this feature and the original goal of the introduction of this keyword. Those who know a bit of assembly know that pushing a lot of parameters on the stack to call function can cost more instructions than the function holds.
```

Comment:
[Suggestion] You need some words to point to "instructions" in clause "than the function holds"